src/meili/map_matcher.cc

```diff
@@ -216,8 +217,9 @@ FindMatchResult(const MapMatching& mapmatching,
     const auto& next_state = mapmatching.state(next_stateid);
     // It must stay on the first edge of the route
     for (auto label = state.RouteBegin(next_state); label != state.RouteEnd(); label++) {
-      if (label->edgeid.Is_Valid()) {
-        edgeid = label->edgeid;
+      if (label->edgeid().Is_Valid()) {
```

Comment:
I need to rephrase it:

We need the dummy label to tell if the route is empty or is not found. If the origin pathlocation and a destination pathlocation are at the same node, then the route between is empty. With the use of the dummy label, we can tell: If the `RouteBegin() == RouteEnd()` then the route is not found, otherwise, there is a route, even if the route is empty (i.e. `RouteBegin()-> predecessor()` is invalid)